@xen-orchestra/proxy/src/app/mixins/backups/_RemoteAdapter.js

```diff
@@ -371,23 +371,45 @@ export class RemoteAdapter {
     const backups = []
 
     try {
-      const files = await handler.list(`${BACKUP_DIR}/${vmUuid}`, {
-        filter: isMetadataFile,
-        prependDir: true,
-      })
-      await asyncMap(files, async file => {
+      const files = (
+        await handler.list(`${BACKUP_DIR}/${vmUuid}`, {
+          filter: isMetadataFile,
+          prependDir: true,
+        })
+      ).sort()
+
+      const chainSizeByVdi = {}
+      for (const file of files) {
         try {
           const metadata = await this.readVmBackupMetadata(file)
           if (predicate === undefined || predicate(metadata)) {
             // inject an id usable by importVmBackupNg()
             metadata.id = metadata._filename
 
+            // a backup size is a sum of all its VHDs' size
+            // a VHD size is a sum of the VHDs' chain size
```

Comment:
It was not what we talked about, I said that simply reporting the size of this backup, not the consolidated size, would be enough :slightly_smiling_face:

Reply:
Why not simply do like full backups by adding the size on the backup execution?

Reply:
I agree, they should be handled similarly. 👍